### PR TITLE
Ux 131 dropdown api change

### PR DESCRIPTION
**Pull request by tristanjasper**
### Purpose 🚀
Simplify the trigger Api for the dropdownComponent

### Notes ✏️

Secondary fixes include

- Added the edge property on dropdown component for alignment to the trigger
- Cleanup of the story knobs 
- Changed the Dropdown.Trigger component api to use `buttonType` prop allowing us to pass any content and set the button appearance.

### Updates 📦
- [x ] MAJOR (breaking) change to _these packages_
- [ ] MINOR (backward compatible) change to _these packages_ 
- [ ] PATCH (big fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/your-branch-name

### Screenshots 📸
_optional but highly recommended_

### References 🔗
https://github.com/acl-services/paprika/issues/131_

### CODE REVIEW

File: packages/DropdownMenu/src/components/Trigger/Trigger.js [hunk 1]
```diff
@@ -1,38 +1,55 @@
...
+  simple: Button,
+};
+
+ButtonTypes.ALL = Object.values(ButtonTypes);
+
 const propTypes = {
   children: PropTypes.node,
-  icon: PropTypes.node,
-  isOpen: PropTypes.bool.isRequired,
+  buttonType: PropTypes.oneOf(ButtonTypes.ALL),
```
Comment (tristanjasper) [inline packages/DropdownMenu/src/components/Trigger/Trigger.js]: changing the api of the trigger component to take a `buttonType` prop. This makes it easy for us to pass any content we like, and just change a prop to say it should show the content in a button as default or as a raw button for example which doesnt' show the button styles

File: packages/DropdownMenu/src/DropdownMenu.js [hunk 2]
```diff
@@ -80,6 +76,8 @@ const DropdownMenu = props => {
       const confirmationComponent = renderConfirmation(handleCloseMenu);
       return React.cloneElement(confirmationComponent, {
         align,
+        edge,
```
Comment (tristanjasper) [inline packages/DropdownMenu/src/DropdownMenu.js]: ensuring the confirmation component uses the same edge value as the rest of the items in the dropdown

File: packages/DropdownMenu/stories/examples/Showcase.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import { Rule, Tagline } from "storybook/assets/styles/common.styles";
-import { select } from "@storybook/addon-knobs";
+import { select, text } from "@storybook/addon-knobs";
+import { AlignTypes } from "@paprika/helpers/lib/customPropTypes";
```
Comment (nahumzs) [inline packages/DropdownMenu/stories/examples/Showcase.js]: `out of the scope for this PR`

I think we should start creating a types file on helper package and have something like:

```
// types.js
export default {
  left: "LEFT",
 right: "RIGHT",
 xsmall: "XSMALL",
 large: "LARGE",
 ...etc
}
```

```js
import types from "@paprika/helpers/lib/types";

const defaultProps: {
  size: types.xsmall
}

function Something(props) {
   const {size} = props;
   if (size === types.xsmall) {
        render ""xsmall"
   }

   return "something else"
}
```

cc @mikrotron  what do you think?

File: packages/DropdownMenu/stories/examples/DropdownMenuDividersExample.js [hunk 4]
```diff
@@ -3,14 +3,8 @@ import DropdownMenu from "../../src";
...
-      align="bottom"
-      renderTrigger={({ isOpen, handleOpenMenu }) => (
-        <DropdownMenu.Trigger isOpen={isOpen} onOpenMenu={handleOpenMenu}>
-          Trigger
-        </DropdownMenu.Trigger>
-      )}
-    >
+    <DropdownMenu align="bottom">
+      <DropdownMenu.Trigger>Trigger</DropdownMenu.Trigger>
       <DropdownMenu.Item onClick={() => {}}>Item 1</DropdownMenu.Item>
```
Comment (nahumzs) [inline packages/DropdownMenu/stories/examples/DropdownMenuDividersExample.js]: In an unrelated but taking advantage of this PR.

I don't think onClick should be required on DropDown.Item, 
what bout if you just want to display a `<hr />` ? do we have a `<DropDown.Divider />`?

**Comment by nahumzs** [comment]
Thank you tristan :)

File: packages/DropdownMenu/stories/examples/Showcase.js (hunk 3 above)
Comment (mikrotron) [reply packages/DropdownMenu/stories/examples/Showcase.js]: Yeah, I think having a `types` packages is a good idea.  I don't think we should export the t-shirt sizes as the default export, but that's a small implementation detail.  

Another possibility is to have things like `types`, `helpers`, maybe even `tokens` and `stylers` all bundled together in a `core` package...

Comment (nahumzs) [reply packages/DropdownMenu/stories/examples/Showcase.js]: shouldn't all of them be part of the helper package?

File: packages/DropdownMenu/stories/examples/DropdownMenuDividersExample.js (hunk 4 above)
Comment (tristanjasper) [reply packages/DropdownMenu/stories/examples/DropdownMenuDividersExample.js]: Yes we have a Dropdown.Divider. Currently the `onClick` isn't required either. We have `Dropdown.Item` that have an `onClick` prop, but also `Dropdown.LinkItem` that doens't have an `onClick` prop as it is rendered as an `a` tag

**Pull request merged**